core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueServiceImpl.java

```diff
@@ -175,29 +181,67 @@ public CompletableFuture<CheckStatusChangePermittedResponse> checkStatusChangePe
     }
     var issueKey = params.getIssueKey();
     var serverApi = serverApiOpt.get();
+
+
+
     return asUUID(issueKey)
       .flatMap(localOnlyIssueRepository::findByKey)
       .map(r -> {
+        // For anticipated issues we currently don't get the information from SonarQube (as there is no web API
+        // endpoint) regarding the available transitions. SonarCloud doesn't provide it currently anyway.
+        // -> Here we just rely on the SonarQube version alone. God have mercy with the poor soul who has to change it
+        //    once SonarCloud is supporting anticipated transitions or SonarQube wants to change it once again :>
```

Comment:
I don't believe that calling to a higher being is necessary here 😉; from experience, a joke in a comment is little comfort when trying to understand why a change was made / how a bug happens.

If the code is not clear about its intention, and if the goal is to express empathy with the future reader, I would rather change the code to make it clearer, if possible.